### ISSUE TITLE
Linux: segfault in "GetCurrentWindowTitle" function

### ISSUE THREAD

**Issue by bitc** [WarmUpTill/SceneSwitcher#158]
Hello, I am using SceneSwitcher 1.11.1 binary release for linux downloaded from GitHub.

When I open up the plugin and click "Start" I get a crash immediately, no matter how I configure it (even if I do no configuration).

Here is the backtrace from gdb:

```
Thread 19 "QThread" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff84dac700 (LWP 8117)]
0x00007fff8c74c4e6 in GetCurrentWindowTitle (title="") at /home/runner/work/SceneSwitcher/SceneSwitcher/UI/frontend-plugins/SceneSwitcher/src/linux/advanced-scene-switcher-nix.cpp:209
209     /home/runner/work/SceneSwitcher/SceneSwitcher/UI/frontend-plugins/SceneSwitcher/src/linux/advanced-scene-switcher-nix.cpp: No such file or directory.
(gdb) bt
#0  0x00007fff8c74c4e6 in GetCurrentWindowTitle(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (title="") at /home/runner/work/SceneSwitcher/SceneSwitcher/UI/frontend-plugins/SceneSwitcher/src/linux/advanced-scene-switcher-nix.cpp:209
#1  0x00007fff8c72b8d5 in SwitcherData::checkPause() (this=this@entry=0x55555662fda0) at /home/runner/work/SceneSwitcher/SceneSwitcher/UI/frontend-plugins/SceneSwitcher/src/switch-pause.cpp:176
#2  0x00007fff8c64f744 in SwitcherData::Thread() (this=0x55555662fda0) at /home/runner/work/SceneSwitcher/SceneSwitcher/UI/frontend-plugins/SceneSwitcher/src/advanced-scene-switcher.cpp:512
#3  0x00007ffff397517d in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#4  0x00007ffff2d726db in start_thread (arg=0x7fff84dac700) at pthread_create.c:463
#5  0x00007ffff2a9b71f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) info args
title = ""
(gdb) info locals
active = 398
actualType = 0
format = 0
num = 0
bytes = 0
data = 0x0
name = <optimized out>
rootWin = <optimized out>
xstatus = <optimized out>
status = 0
text = {value = 0x55555662fdcc "", encoding = 140737267601021, format = 1457904528, nitems = 0}
(gdb) 
```

Here is the line where it is crashing:

<https://github.com/WarmUpTill/SceneSwitcher/blob/10e1c7b3a30044f61da799fc4386132e157fe9e2/src/linux/advanced-scene-switcher-nix.cpp#L209>

You can see that the `data` pointer is a null pointer and that is why it is crashing.

I am using xmonad window manager.

Thank you

**Comment by WarmUpTill** [comment]
Thanks for reporting the issue.
Just to clarify: Is this a new crash or did you also observe this behavior in the past?

**Comment by bitc** [comment]
This is the first time I have installed and am using "SceneSwitcher". But my guess from looking at the C++ code here is that this crash exists for a long long time.

The good news: I configured my window manager (xmonad) to enable "Extended Window Manager Hints" (ewmh) and now I don't get a crash, and everything works perfectly. I followed the instructions in the last post here: <https://obsproject.com/forum/threads/tiling-window-managers-xcomposite-no-windows-found.34708/>
